services/motion/builtin/move_request.go

```diff
@@ -200,70 +332,54 @@ func (mr *moveRequest) obstaclesIntersectPlan(
 		plan = append(plan, input)
 	}
 
-	for visSrvc, cameraNames := range mr.obstacleDetectors {
-		for _, camName := range cameraNames {
-			mr.logger.Debugf(
-				"proceeding to get detections from vision service: %s with camera: %s",
-				visSrvc.Name().ShortName(),
-				camName.ShortName(),
-			)
-
-			// get detections from vision service
-			detections, err := visSrvc.GetObjectPointClouds(ctx, camName.Name, nil)
-			if err != nil {
-				return state.ExecuteResponse{}, err
-			}
+	// get the already defined geometries of worldstate
+	existingGifs, err := mr.planRequest.WorldState.ObstaclesInWorldFrame(
+		mr.planRequest.FrameSystem,
+		referenceframe.StartPositions(mr.planRequest.FrameSystem),
```

Comment:
Continuing from old thread that got nuked due to PR changes:

If the camera is mounted on something InputEnabled that isn't the base, then that input needs to be known in order to properly calculate the pose of the obstacle.

Furthermore, if that InputEnabled thing has moved since this moveRequest was initialized (due to some other non-motion call for example), then we can't just get current inputs; we need the original input to place that thing in its original position. Thus the CurrentInputs from the start should be cached and used.

Fortunately this already exists; `mr.planRequest.StartConfiguration`